src/main/java/dev/openfeature/sdk/EventProvider.java

```diff
@@ -16,21 +18,56 @@
  */
 public abstract class EventProvider implements FeatureProvider {
 
+    private ProviderState providerState = ProviderState.NOT_READY;
+
     /**
      * {@inheritDoc}
      */
     @Override
-    public abstract ProviderState getState();
+    public final ProviderState getState() {
+        return providerState;
+    }
+
+    @Override
+    public final void initialize(EvaluationContext evaluationContext) throws Exception {
+        try {
+            doInitialization(evaluationContext);
+            providerState = ProviderState.READY;
+        } catch (OpenFeatureError openFeatureError) {
+            if (ErrorCode.PROVIDER_FATAL.equals(openFeatureError.getErrorCode())) {
+                providerState = ProviderState.FATAL;
+            } else {
+                providerState = ProviderState.ERROR;
+            }
+            throw openFeatureError;
+        } catch (Exception e) {
+            providerState = ProviderState.ERROR;
+            throw new GeneralError(e);
+        }
+    }
+
+    protected void doInitialization(EvaluationContext evaluationContext) throws Exception {
+        // Intentionally left blank, to be implemented by inheritors
+    }
+
+    @Override
+    public final void shutdown() {
+        providerState = ProviderState.NOT_READY;
+        doShutdown();
```

Comment:
why this split is needed ?
without it, provider can implement shutdown() same as today and calling _super.shutdown()_ in it.
providers which were implemented this way, can continue to work as is.
similar with initialize().